packages/api-augment/tsconfig.json

```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "rootDir": "src",
     "outDir": "build",
+    "module": "esnext",
```

Comment:
What is the reason to use esnext, the experimental features of the next ECMAScript? Change all **"module"** and **"target"** to "es2017" and see if they work.

From chatGPT:
If you're looking for a more compatible and widely supported way of setting the target in your `tsconfig.json`, you can choose an ECMAScript version that aligns with your target environment. Here are some alternatives:

1. `"target": "es5"`: This targets ECMAScript 5, which is widely supported across all modern browsers and JavaScript engines. It provides good compatibility for most environments.

2. `"target": "es2015"` or `"target": "es6"`: These targets ECMAScript 2015 (ES6), which introduced significant language enhancements. While not supported in older browsers like Internet Explorer, it is well-supported in modern browsers and Node.js.

3. `"target": "es2017"` or `"target": "es7"`: These targets ECMAScript 2017, which includes additional language features and improvements. It is supported by most modern browsers and environments.

4. `"target": "es2019"` or `"target": "es9"`: These targets ECMAScript 2019, which includes further language enhancements. It is widely supported by modern browsers and environments.

By choosing one of these alternatives, you can ensure broader compatibility and support for your compiled TypeScript code. It allows you to target a specific ECMAScript version that aligns with your deployment requirements and the browsers or environments you intend to support. Remember to consider your target audience and the minimum requirements for your project to make an informed decision about the appropriate target version.

Reply:
Yes, I referenced the configuration from the Moonbeam project.

https://github.com/PureStake/moonbeam/blob/82035b77cd48c2fffb44907ce1501b2128117213/typescript-api/tsconfig.base.json#L4

I haven't delved too deeply into these configurations, but they work correctly.

Reply:
`module: "es2017"` is causing an error.
```
TSError: ⨯ Unable to compile TypeScript:
error TS6046: Argument for '--module' option must be: 'none', 'commonjs', 'amd', 'system', 'umd', 'es6', 'es2015', 'es2020', 'es2022', 'esnext', 'node16', 'nodenext'.
```

However, using `module: "es2015"` works.

We can choose either `esnext` or `es2015`.

Reply:
okay, module: es2017 didn’t work. Let’s keep it as is, `"module": "esnext"`